core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultDriverConfigLoader.java

```diff
@@ -140,47 +155,59 @@ private SingleThreaded(InternalDriverContext context) {
               .getDefaultProfile()
               .getDuration(DefaultDriverOption.CONFIG_RELOAD_INTERVAL);
 
-      forceLoadListenerKey =
-          this.eventBus.register(
-              ForceReloadConfigEvent.class, e -> RunOrSchedule.on(adminExecutor, this::reload));
-
-      RunOrSchedule.on(adminExecutor, this::scheduleReloadTask);
+      RunOrSchedule.on(adminExecutor, this::schedulePeriodicReload);
     }
 
-    private void scheduleReloadTask() {
+    private void schedulePeriodicReload() {
       assert adminExecutor.inEventLoop();
       // Cancel any previously running task
-      if (reloadFuture != null) {
-        reloadFuture.cancel(false);
+      if (periodicTaskHandle != null) {
+        periodicTaskHandle.cancel(false);
       }
       if (reloadInterval.isZero()) {
         LOG.debug("[{}] Reload interval is 0, disabling periodic reloading", logPrefix);
       } else {
         LOG.debug("[{}] Scheduling periodic reloading with interval {}", logPrefix, reloadInterval);
-        reloadFuture =
+        periodicTaskHandle =
             adminExecutor.scheduleAtFixedRate(
-                this::reload,
+                () -> reload(null),
                 reloadInterval.toNanos(),
                 reloadInterval.toNanos(),
                 TimeUnit.NANOSECONDS);
       }
     }
 
-    private void reload() {
+    private void reload(CompletableFuture<Void> reloadedFuture) {
       assert adminExecutor.inEventLoop();
       if (closeWasCalled) {
+        if (reloadedFuture != null) {
+          reloadedFuture.completeExceptionally(new IllegalStateException("session is closing"));
+        }
         return;
       }
-      if (driverConfig.reload(configSupplier.get())) {
-        LOG.info("[{}] Detected a configuration change", logPrefix);
-        eventBus.fire(ConfigChangeEvent.INSTANCE);
-        Duration newReloadInterval = config.getDuration(DefaultDriverOption.CONFIG_RELOAD_INTERVAL);
-        if (!newReloadInterval.equals(reloadInterval)) {
-          reloadInterval = newReloadInterval;
-          scheduleReloadTask();
+      try {
+        if (driverConfig.reload(configSupplier.get())) {
```

Comment:
I think that this method `dirverConfig.reload` could be named to smth like: `hasConfigChanged()` `checkIfConfigChange`. Currently, the name `reload()` is not telling the caller that it is checking if only change. From the name, I would guess that it triggers reload

Reply:
It also reloads actually: `TypesafeDriverConfig` is mutable, and if the configuration has changed it will reflect the new values. I feel like `hasConfigChanged()` would make it sound like it only checks.
Do you think adding more explanations in the method's javadocs would be sufficient? Or maybe also rename the method to `updateWith(Config)`.

Reply:
Docs is ok, when I looked into it the behavior was clear.
Maybe `relaoadAndCheckIfChanged()`?

Reply:
That's a bit of a mouthful... The problem if we try to express everything the method does is that this is endless, for example it also logs a warning if a profile was added at runtime.

I think `reload` will be fine. If there's any doubt as to what the result represents, the javadoc is only what shortcut/jump away.